File: src/main/java/nsu/auth/JwtUtils.java
package nsu.auth;

import io.jsonwebtoken.Claims;
import nsu.auth.JwtAuthentication;
import nsu.auth.Roles;

import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

//@NoArgsConstructor(access = AccessLevel.PRIVATE)

public final class JwtUtils {

    public static JwtAuthentication generate(Claims claims) {
        final JwtAuthentication jwtInfoToken = new JwtAuthentication();
        jwtInfoToken.setRoles(getRoles(claims));
        jwtInfoToken.setFullname(claims.get("fullName", String.class));
        jwtInfoToken.setEmail(claims.getSubject());
        return jwtInfoToken;
    }

    private static Set<Roles> getRoles(Claims claims) {
        final List<String> roles = claims.get("roles", List.class);
        return roles.stream()
                .map(Roles::valueOf)
                .collect(Collectors.toSet());
    }

}

File: src/main/kotlin/nsu/auth/jwt/JwtUtils.kt
package nsu.auth.jwt

import io.jsonwebtoken.Claims
import nsu.auth.entity.Roles
import java.util.stream.Collectors

//@NoArgsConstructor(access = AccessLevel.PRIVATE)
object JwtUtils {
    fun generate(claims: Claims): JwtAuthentication {
        val jwtInfoToken = JwtAuthentication()
        jwtInfoToken.roles = getRoles(claims)
        jwtInfoToken.fullname = claims.get("fullName", String::class.java)
        jwtInfoToken.email = claims.subject
        return jwtInfoToken
    }

    private fun getRoles(claims: Claims): Set<Roles?> {
        val roles: List<String> = claims.get("roles", List::class.java) as List<String>
        return roles.stream().map { role: String? -> Roles.valueOf(role!!) }.collect(Collectors.toSet())
    }
}
